src/neo-vm/ExecutionEngine.cs

```diff
@@ -764,26 +1277,45 @@ private void ExecuteOp(OpCode opcode, ExecutionContext context)
                             switch (context.EvaluationStack.Pop())
                             {
                                 case VMArray array:
-                                    int index = (int)key.GetBigInteger();
-                                    if (index < 0 || index >= array.Count)
                                     {
-                                        State |= VMState.FAULT;
-                                        return;
+                                        int index = (int)key.GetBigInteger();
+                                        if (index < 0 || index >= array.Count)
+                                        {
+                                            State |= VMState.FAULT;
+                                            return;
+                                        }
+                                        array[index] = value;
+                                        break;
                                     }
-                                    array[index] = value;
-                                    break;
                                 case Map map:
-                                    map[key] = value;
-                                    break;
+                                    {
+                                        if (!map.ContainsKey(key) && !CheckArraySize(map.Count + 1))
```

Comment:
Isn't this an || ?

Reply:
if is contained, the stack count is the same

Reply:
oh man!! correct... if not contained it adds one... so much optimization for my simple head :)